pkg/controller/servicebindingrequest/retriever_test.go

```diff
@@ -102,3 +102,44 @@ func TestRetriever(t *testing.T) {
 		assert.Nil(t, err)
 	})
 }
+
+func TestRetrieverNestedCRDKey(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	var retriever *Retriever
+
+	ns := "testing"
+	crName := "db-testing"
+
+	crdDescription := mocks.CRDDescriptionMock()
+	cr := mocks.NestedDatabaseCRMock(ns, crName)
+
+	genericCR, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&cr)
+	require.Nil(t, err)
+
+	plan := &Plan{
+		Ns:             ns,
+		Name:           "retriever",
+		CRDDescription: &crdDescription,
+		CR:             &ustrv1.Unstructured{Object: genericCR},
+	}
+
+	dbSecret := mocks.SecretMock(ns, "db-credentials")
+	objs := []runtime.Object{&dbSecret}
+	fakeClient := fake.NewFakeClient(objs...)
+
+	retriever = NewRetriever(context.TODO(), fakeClient, plan)
+	require.NotNil(t, retriever)
+
+	t.Run("Second level", func(t *testing.T) {
```

Comment:
Is it worth adding an edge case here - where the path does not include a "." char?

Or - maybe a test of an invalid path (that should raise an error) such as "first..second"?

Reply:
There is a test without any dot in line 50.

Added test for two dot scenario: b0f80a0c68a422e9d941f3b62f2fb9eb48c8f93a